### ISSUE TITLE
When the device loses connection for a short period of time triggers "receive loop has terminated" and never reloads

### ISSUE THREAD

**Issue by mihsu81** [make-all/tuya-local#912]
This form is for reporting bugs.  If you have encountered a bug, you should
generally have some log messages or device diagnostics to show something is
going wrong.  If your problem is more of a question or a request for help,
then [Discussions](https://github.com/make-all/tuya-local/discussions) is
probably a more suitable venue.

**Describe the bug**
If the device (in my case a dehumidifier D12 ECO 2.0 (gcthce4ejb8nzgow)) loses connection to WiFi for whatever reason, I get some errors in HA and the integration never tries to reestablish connection to it.

**To Reproduce**
Reboot WiFi, or move the device away from the router/AP and wait a bit until the issue appears.

**Expected behavior**
The integration should check periodically if the device becomes available again and retry creating the receive loop.

**Additional context**

If the bug involves a device, then please include device diagnostics from
Settings / Devices & Services / Tuya Local (your device) / "1 device"
[config_entry-tuya_local-3ac6f091d87c2d54f17d7d481712c8f6.json.txt](https://github.com/make-all/tuya-local/files/12103453/config_entry-tuya_local-3ac6f091d87c2d54f17d7d481712c8f6.json.txt)

Please check if there are messages from Tuya Local in the Home
Assistant log (Settings / System / Logs), and include those too.
```
This error originated from a custom integration.

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:207
Integration: Tuya Local (documentation, issues)
First occurred: July 18, 2023 at 23:29:45 (2 occurrences)
Last logged: 07:43:25

Bathroom 1 Dehumidifier receive loop has terminated
```
```
This error originated from a custom integration.

Logger: custom_components.tuya_local.device
Source: custom_components/tuya_local/device.py:184
Integration: Tuya Local (documentation, issues)
First occurred: July 18, 2023 at 23:28:38 (2 occurrences)
Last logged: 07:40:15

Failed to refresh device state for Bathroom 1 Dehumidifier
```

**Comment by make-all** [comment]
Do you know what protocol version your device is using?

**Comment by make-all** [comment]
It may help with the recovery to have it fixed. It can take up to 15 minutes to cycle through non-working protocols, and the wrong protocols may be causing the device to go offline for long enough that it misses the 3.3 attempts.

**Comment by mihsu81** [comment]
Setting a fixed protocol (3.3 in this case) solved the issue. Thank you.
